version: "3"

services:
  api1:
    environment:
      DATABASE_URL: postgres://postgres:postgres@db:5432/rinhapython
      PORT: 8001
    build:
      context: .
    volumes:
      - .:/app:z
    depends_on:
      db: 
        condition: service_healthy
    network_mode: "host"
    deploy:
      resources:
        limits:
          cpus: '0.25'
          memory: '446MB'

  api2:
    environment:
      DATABASE_URL: postgres://postgres:postgres@db:5432/rinhapython
      PORT: 8002
    build:
      context: .
    volumes:
      - .:/app:z
    depends_on:
      db: 
        condition: service_healthy
    network_mode: "host"
    deploy:
      resources:
        limits:
          cpus: '0.25'
          memory: '446MB'

  nginx:
    image: nginx:1.25.3
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
    depends_on:
      - api1
      - api2
    network_mode: "host"
    deploy:
        resources:
          limits:
            cpus: '0.5'
            memory: '128MB'

  db:
    image: "postgres:12-bullseye"
    hostname: postgres
    volumes:
      - postgresql_data:/var/lib/postgres/data
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 10s
      timeout: 5s
      retries: 5
    environment:
      - POSTGRES_DB=rinhapython
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - PGPASSWORD=postgres
    network_mode: "host"
    deploy:
        resources:
          limits:
            cpus: '0.5'
            memory: '512MB'

volumes:
  postgresql_data:
  init.sql:
